src/addon/esmpy/src/esmpy/interface/cbindings.py

```diff
@@ -218,16 +218,22 @@ def ESMP_Initialize(logkind = constants.LogKind.MULTI):
 _ESMF.ESMC_Finalize.restype = ct.c_int
 _ESMF.ESMC_Finalize.argtypes = []
 
-def ESMP_Finalize():
+def ESMP_Finalize(endFlag = constants.EndAction.NORMAL):
     """
     Preconditions: ESMF has been initialized.
     Postconditions: ESMF has been finalized, all heap memory has been
-                    released, and all MPI states have been cleaned up.
-                    This method can only be called once per execution,
-                    and must be preceded by one and only one call to
-                    ESMP_Initialize(). \n
+                    released, and, if 'keepMpi' is False, all MPI states
```

Comment:
Needs to be changed to say something about endFlag equaling Normal.

Reply:
I think this was in reference to an older commit, but I revised the docstring again to be explicit about MPI cleanup only when endFlag is set to `NORMAL` (and not necessarily when it's set to something other than `KEEP_MPI`). Fixed in [0f7b9b4](https://github.com/esmf-org/esmf/pull/234/commits/0f7b9b4a215f8648c472a48d3a262ada19b09bab).

Reply:
Looks good! Thanks!